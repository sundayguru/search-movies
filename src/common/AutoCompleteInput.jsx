import React from 'react';
import TextInput from 'react-autocomplete-input';
import 'react-autocomplete-input/dist/bundle.css';


class AutoCompleteInput extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      labelField: props.labelField ? props.labelField : 'title', 
      trigger: props.trigger ? props.trigger : '',
      options: []
    }
    this.onRequestOptions = this.onRequestOptions.bind(this)
  }

  onRequestOptions() {
    this.setState({options: this.props.items.map((item) => item[this.state.labelField].toLowerCase())})
  }

  render() {
    const { options, trigger } = this.state;

    return (
      <TextInput 
        trigger={trigger}
        onRequestOptions={this.onRequestOptions}
        requestOnlyIfNoOptions={true} 
        options={options}
        onChange={this.props.onChange}
        className={this.props.className}
        placeholder={this.props.placeholder}
        Component="input"
      />
    )
  }
}

export default AutoCompleteInput;