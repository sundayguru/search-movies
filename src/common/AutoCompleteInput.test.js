import React from 'react'
import AutoCompleteInput from './AutoCompleteInput'
import renderer from 'react-test-renderer'


test('<SearchBox />', () => {
  const component = renderer.create(
    <AutoCompleteInput items={['a', 'b']} />,
  );

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
