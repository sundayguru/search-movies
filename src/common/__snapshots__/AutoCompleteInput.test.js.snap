// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> 1`] = `
<span>
  <input
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
</span>
`;
